src/client/containers/CandidateList/CandidateList.js

```diff
@@ -0,0 +1,62 @@
+import React, { useState } from 'react';
+import { useCandidates } from '../UseHooks/useCandidates';
+import { useUpdateBallots } from '../UseHooks/useUpdateBallots';
+import { CardItemDecorator } from '../../components/CandidateCard/CandidateCardItem.component';
+import { candidateCardSorting } from '../../components/CandidateCard/CandidateCardSorting';
+import {
+  DragAndSortAdapter,
+  SortableItem,
+} from '../DragAndSortAdapter/DragAndSortAdapter';
+import { onDragEnd } from '../DragAndSortAdapter/OnDragEnd';
+import PropTypes from 'prop-types';
+
+export default function CandidateList({ userId, boardId }) {
+  const { candidates, error } = useCandidates(userId, boardId);
+  const [updatedCandidates, setUpdatedCandidates] = useState(candidates);
+  const { error: updateError } = useUpdateBallots(
+    userId,
+    boardId,
+    updatedCandidates,
+  );
+
+  const onClick = () => {
+    console.log('clicked');
+  };
+
+  return (
+    <div className="CandidateCard-component">
+      {error && updateError ? (
+        <h2 className="showups">{error || updateError}</h2>
+      ) : (
+        <DragAndSortAdapter
+          onDragEndHandler={onDragEnd(
+            setUpdatedCandidates,
+            candidateCardSorting,
+          )}
+          items={updatedCandidates}
+        >
+          {updatedCandidates.map((candidate) => {
+            return (
+              <SortableItem
+                key={candidate.candidateId}
+                id={candidate.candidateId}
+              >
+                <CardItemDecorator
+                  colorVariant="primary-color"
+                  candidateName={candidate.name}
+                  displayDeleteIcon="visible"
+                  onClick={onClick}
+                />
+              </SortableItem>
+            );
+          })}
+        </DragAndSortAdapter>
```

Comment:
I guess these components are build over the use of `candidate.id`. This drag and drop functionality is not working properly even if change id to candidateId.

Reply:
The candidate id is whatever the candidate id is called when you get the response when getting the candidates.

Reply:
Sorry you are right. You have to change it in candidateCardSorting also

```
export function candidateCardSorting(candidates, activeId, overId) {
  return {
    oldIndex: candidates.findIndex((candidate) => candidate.id === activeId),
    newIndex: candidates.findIndex((candidate) => candidate.id === overId),
  };
}
```